### ISSUE TITLE
[spec failure]  standard_couch removes and returns a value from the backing store via delete if it exists

### ISSUE THREAD

**Issue by asppsa** [minad/moneta#125]
~~~
1) standard_couch removes and returns a value from the backing store via delete if it exists
     Failure/Error: raise "HTTP error #{response.status}" unless response.status == 200
     RuntimeError:
       failure: 39 tests, on:
       ["\x15\x16\x7F\x14\e\x7F", {"kaWKN0"=>[884513996833579312, {"Q6Iwdm"=>492500693963998093, "8cBcQD"=>-1475025662084525079, "0eBhOp"=>2175057818714474246, "2Ip5bg"=>1735665960320244152, "nsXbw4"=>-99268057728669189, "AGC02v"=>1247045661452617391}], "vQ5532"=>["dTikYX", -135008491646992724], "fXbgHn"=>[1738560228845555672, "NDylQl"], "OZDN6R"=>[{"AUHaqr"=>1223920004966990961, "OPtuxd"=>92759660056523433, "OuWqKv"=>-1198474969971489881, "lG2lk8"=>-1122728737065436400, "kb21y0"=>2274735426143285775, "b0rnN2"=>2035424328192694803}, -1152895400508308124], "ByKmN0"=>["hbT6VV", {"HtMBzM"=>-541705218268898711, "TZReo2"=>2116402108589333537, "Ub9Dux"=>1502573750790742097, "K4ejW6"=>1688863875631213936, "ue5R1P"=>1950390870367352981, "Du6Ku3"=>825635776516655507}], "dUw74T"=>[-134573552179028077, {"WrWK6Z"=>-1490425510792530903, "CCvD5C"=>-73979366661919350, "9hLmZ4"=>-1062211304465984183, "RdyjuG"=>-1545816492507490264, "yIQ8Qk"=>-908451564065361938, "wSTAey"=>-1806914659334026386}]}]
       HTTP error 400
     Shared Example Group: :store called from ./spec/helper.rb:241
     # ./lib/moneta/adapters/couch.rb:66:in `delete'
     # ./lib/moneta/transformer.rb:139:in `delete'
     # ./lib/moneta/proxy.rb:75:in `delete'
     # ./lib/moneta/wrapper.rb:22:in `block in delete'
     # ./lib/moneta/logger.rb:66:in `wrap'
     # ./lib/moneta/wrapper.rb:22:in `delete'
     # ./spec/features/store.rb:40:in `block (3 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.1.0/gems/rantly-1.1.0/lib/rantly/property.rb:33:in `call'
     # ./vendor/bundle/ruby/2.1.0/gems/rantly-1.1.0/lib/rantly/property.rb:33:in `block in check'
     # ./vendor/bundle/ruby/2.1.0/gems/rantly-1.1.0/lib/rantly/generator.rb:83:in `call'
     # ./vendor/bundle/ruby/2.1.0/gems/rantly-1.1.0/lib/rantly/generator.rb:83:in `generate'
     # ./vendor/bundle/ruby/2.1.0/gems/rantly-1.1.0/lib/rantly/property.rb:31:in `check'
     # ./spec/features/store.rb:38:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # RuntimeError:
     #   HTTP error 400
     #   ./lib/moneta/adapters/couch.rb:66:in `delete'
~~~

**Comment by asppsa** [comment]
This issue is due to the fact that the base64-encoded version of `"\x15\x16\x7F\x14\e\x7F"` is `"FRZ/FBt/"`, which has slashes in it.  CouchDB seems to happily accept GET and PUT requests to this URL, although actually the doc is stored at "FRZ" (so that's another issue with this).  DELETE requests to such a URL trigger a 400 error though.

The solution may be to add URL-safe base64 as a transformer, and use that for CouchDB keys.

**Comment by asppsa** [comment]
Fixed in ff2b398340586d81c944f9f341711d732cfa595b
